### PR TITLE
feat: add sidemenu component

### PR DESCRIPTION
**Pull request by gary-van-woerkens**

### CODE REVIEW

File: src/SideMenu.tsx [hunk 1]
```diff
@@ -38,7 +44,14 @@ export const SideMenu = memo(
                 ref={ref}
                 style={style}
                 aria-labelledby="fr-sidemenu-title"
-                className={cx(fr.cx("fr-sidemenu"), className)}
+                className={cx(
+                    fr.cx("fr-sidemenu", {
+                        "fr-sidemenu--sticky": sticky === true,
```
Comment (garronej) [inline src/SideMenu.tsx:49]:
```suggestion
                        "fr-sidemenu--sticky": sticky,
```

Comment (gary-van-woerkens) [reply src/SideMenu.tsx:49]: If we want a sticky aligned right menu, we have to mix both the two classes `fr-sidemenu--right` and `fr-sidemenu--sticky-full-height` ?

Comment (gary-van-woerkens) [reply src/SideMenu.tsx:49]: I confirm that we don't need both classes at the same time to get the sticky menu working.
So stop me if I'm wrong but if we don't test the `sticky === true` we might have `fr-sidemenu--sticky` and `fr-sidemenu--sticky-full-height` present at the same time.

**Comment by gary-van-woerkens** [comment]
Pipeline is failing here:
https://github.com/codegouvfr/react-dsfr/actions/runs/4440161762/jobs/7793612144?pr=79#step:5:10

But I don't manage to reproduce the error locally 😕

**Comment by gary-van-woerkens** [comment]
Pipeline problem may be related to what I got locally:

```
~/..../react-dsfr (feature/side-menu ✘)✹ ᐅ git commit -am "multi levels"                        
husky > pre-commit (node v16.13.0)
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up...

→ prettier --write:
src/SideMenu.tsx 215ms
stories/SideMenu.stories.tsx 29ms

→ eslint --fix:
Pages directory cannot be found at /Users/gvanwoerkens/Documents/dnum/react-dsfr/pages or /Users/gvanwoerkens/Documents/dnum/react-dsfr/src/pages. If using a custom path, please configure with the no-html-link-for-pages rule in your eslint config file
[feature/side-menu 8b2f967] multi levels
 3 files changed, 313 insertions(+), 37 deletions(-)
```

**Comment by garronej** [comment]
Hummm 🤔   
Mayber delete all your node_modules and reinstall?

**Comment by gary-van-woerkens** [comment]
Same same...
Also it screams about a file that I didn't touch: `src/Modal.tsx`.

And as far as I see there is the same problem on the `main` branch:
https://github.com/codegouvfr/react-dsfr/actions/runs/4439962563/jobs/7793133989#step:5:10

**Comment by gary-van-woerkens** [comment]
Suggestion... fixing the dependencies versions (removing the `^`) and using **Renovate** or **Dependabot** maybe.

**Comment by garronej** [comment]
> And as far as I see there is the same problem on the main branch:  

Yeah, I merged a PR of someone that made an eddit directly from the github UI. I just had to run `yarn format` to fix it.  

> fixing the dependencies versions (removing the ^)  

It's the role of the `yarn.lock` to pin the dependencies. You'd pin in the package JSON only when there is a known bug in a perticuar dependency and you explicitely don't want tu update.

> and using Renovate or Dependabot maybe.  

It's not usefull in this project. It would just bother us with a ton of notification about dev dependencies updates. 
react-dsfr only has `tsafe` and `gouvfr/dsfr` as dependency and I think the update of `gouvfr/dsfr` shouldn't be automatic.

**Comment by garronej** [comment]
> Same same...  

I just fixed the error, the ignore path was wrong. Thanks for briging it up.  
You can rebase.

File: src/SideMenu.tsx [hunk 2]
```diff
@@ -15,7 +15,10 @@ export type SideMenuProps = {
     align?: "left" | "right";
     items: SideMenuProps.Item[];
     bugerMenuButtonText: ReactNode;
-    sticky?: boolean | "full-height";
+    /** Default: false */
+    sticky?: boolean;
+    /** Default: false, only relevent when sticky */
+    fullHeight: boolean;
```
Comment (garronej) [inline src/SideMenu.tsx:21]:
```suggestion
    fullHeight?: boolean;
```
Sorry

File: src/SideMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import { symToStr } from "tsafe/symToStr";
+import { assert } from "tsafe/assert";
+import type { Equals } from "tsafe";
+import type { RegisteredLinkProps } from "./link";
+import { getLink } from "./link";
+import { fr } from "./fr";
+import { cx } from "./tools/cx";
+
+//https://main--ds-gouv.netlify.app/example/component/sidemenu/
+export type SideMenuProps = {
```
[the remark this answers is not in the document]
Comment (gary-van-woerkens) [reply src/SideMenu.tsx:11]: Not sure that I did it properly...

[the remark this answers is not in the document]
Comment (garronej) [reply src/SideMenu.tsx:11]: LGTM, good job 👍🏻

**Comment by garronej** [comment]
Thank you for your efforts!

**Pull request merged**